crates/rooch-rpc-server/src/service/aggregate_service.rs

```diff
@@ -0,0 +1,302 @@
+                .await?;
+
+            let state = states.pop().flatten().expect("State expected return value");
+            let annotated_coin_store =
+                AnnotatedCoinStore::new_from_annotated_move_value(state.move_value)?;
+
+            let balance_info =
+                BalanceInfo::new_with_default(coin_type, annotated_coin_store.get_coin_value());
+            result.push(Some((None, balance_info)))
+        } else {
+            //TODO If the coin store list exceeds MAX_RESULT_LIMIT_USIZE, consider supporting traverse or pagination
+            let states = self
+                .rpc_service
+                .list_annotated_states(
+                    AccessPath::table_without_keys(coin_store_handle),
+                    cursor,
+                    limit,
+                )
+                .await?;
+
+            let mut data = states
+                .into_iter()
+                .map(|item| {
+                    item.map(|(key, state)| {
+                        let coin_store =
+                            AnnotatedCoinStore::new_from_annotated_move_value(state.move_value)
+                                .expect("AnnotatedCoinStore expected return value");
+
+                        let coin_type =
+                            get_first_ty_as_struct_tag(coin_store.get_coin_struct_type())
+                                .expect("Coin type expected get_first_ty_as_struct_tag succ");
+                        // let inner_coin_info = coin_info_table
+                        //     .get(&coin_type)
+                        //     .expect("Get coin info by coin type expected return value");
+                        let balance_info =
+                            BalanceInfo::new_with_default(coin_type, coin_store.get_coin_value());
+                        (Some(key), balance_info)
+                    })
+                })
+                .collect::<Vec<_>>();
+
+            result.append(&mut data);
+        };
+
+        for (_key, balance_info) in result.iter_mut().flatten() {
+            let coin_info = coin_info_table
+                .get(&balance_info.coin_type)
+                .expect("Get coin info by coin type expected return value");
+            balance_info.symbol = coin_info.symbol.clone();
+            balance_info.decimals = coin_info.decimals;
+        }
+
+        Ok(result)
+    }
+}
+
+impl MoveFunctionCaller for AggregateService {
+    // Use futures::executors::block_on to go from sync -> async
+    // Warning! Possible deadlocks can occur if we try to wait for a future
+    fn call_function(
+        &self,
+        _ctx: &TxContext,
+        function_call: FunctionCall,
+    ) -> Result<FunctionResult> {
+        // let function_result = futures::executor::block_on(async move {
+        // let function_result = futures::executor::block_on(async move {
+        //     self.rpc_service.execute_view_function(function_call).await
+        // })?;
+
+        let function_result =
```

Comment:
We can not use `futures::executor::block_on` here, we need to get the tokio Runtime and use the Runtime block_on. https://tokio.rs/tokio/topics/bridging